src/dmd/typesem.d

```diff
@@ -1341,10 +1341,7 @@ extern(C++) Type typeSemantic(Type t, const ref Loc loc, Scope* sc)
                     }
                 }
 
-                if (fparam.storageClass & (STC.ref_ | STC.lazy_))
-                {
-                }
-                else if (fparam.storageClass & STC.out_)
+                if (fparam.storageClass & STC.out_)
```

Comment:
Is `fparam.storageClass & (STC.ref_ | STC.lazy_)` guaranteed to be false here? Or is it irrelevant?

Reply:
It's rejected. There are a few dead checks in the frontend, and there's this in the parser:
https://github.com/dlang/dmd/blob/04bf3325a3855f622d047d9317dd7a64a70f3956/src/dmd/parse.d#L2965-L2968

Reply:
Good, please add this to the commit message.